param/__init__.py

```diff
@@ -1483,3 +1494,117 @@ def get_range(self):
         return abbreviate_paths(self.path,super(MultiFileSelector, self).get_range())
 
 
+class Date(Number):
+    """
+    Date parameter of datetime type.
+    """
+
+    def __init__(self, default=None, allow_None=False, **kwargs):
+        allow_None = True if default is None else allow_None
+        super(Date, self).__init__(default=default, allow_None=allow_None, **kwargs)
+        self._check_value(default)
+
+    def _check_value(self,val):
+        """
+        Checks that the value is numeric and that it is within the hard
+        bounds; if not, an exception is raised.
+        """
+        if self.allow_None and val is None:
+            return
+
+        if not isinstance(val, dt_types) and not (self.allow_None and val is None):
+            raise ValueError("Date '%s' only takes datetime types."%self._attrib_name)
+
+        self._checkBounds(val)
+
+
+class Color(Parameter):
+    """
+    Color parameter defined as a hex RGB string with an optional #
+    prefix.
+    """
+
+    def __init__(self, default=None, allow_None=False, **kwargs):
+        super(Color, self).__init__(default=default, **kwargs)
+        self._check_value(default)
+
+    def _check_value(self,val):
+        if (self.allow_None and val is None):
+            return
+        if not isinstance(val, String.basestring):
+            raise ValueError("Color '%s' only takes a string value."%self._attrib_name)
+        if not re.match('^#?(([0-9a-fA-F]{2}){3}|([0-9a-fA-F]){3})$', val):
+            raise ValueError("Color '%s' only accepts valid RGB hex codes."
+                             % self._attrib_name)
```

Comment:
Would it be too confusing to accept a numeric triple when it's set, converting it to the RGB hex string required?  The value would change from what the user thought it was; maybe too tricky.